### ISSUE TITLE
UX: Mouse click closes the photo viewer instead of zooming in / opening context menu

### ISSUE THREAD

**Issue by Pheggas** [photoprism/photoprism#1633]
Hello. So i found this bug/weird behaviour. As user i'd like to copy the photo (usually by right clicking on PC) and send it to 3rd party service like Facebook Messenger to share some of my photos with other people. But when i right click, the photo UI just closes and throws me back to the gallery.

**Comment by lastzero** [comment]
Works for me. How did you do that? What operating system & browser version?

**Comment by lastzero** [comment]
We need more information to reproduce this:
- Was it in a specific browser?
- Do you have any plugins installed?
- Does it work with other browsers and/or with plugins disabled?

**Comment by lastzero** [comment]
Works on https://demo.photoprism.org/ using Google Chrome on macOS:

![Screenshot 2021-10-25 at 07 19 01](https://user-images.githubusercontent.com/301686/138638497-f3d34375-2525-4dc8-82fb-d7ae52251b50.png)

**Comment by Pheggas** [comment]
Good to know

- No, it is browser-wide
- If you mean addons then yes. I'm currently using Chrome (95.0.4638.54 64-bit) and have quite a lot addons. Unfortunately, it does this exact thing on any browser i open. 
- No.

It may be caused by it's version tho. As i didn't update photoprism in a while, i have version 210523-b1856b9d-Linux-x86_64

**Comment by lastzero** [comment]
Wouldn't know why, but testing the same version wouldn't hurt either 💎

Can up update the image to see if it's fixed?

**Comment by Pheggas** [comment]
No prob. I'll do it as soon as i'll have more time

**Comment by Pheggas** [comment]
Update: So i get into it, re-pulled image and re-build it. Now i have latest **PhotoPrism® 211018-e200f322-Linux-x86_64** image and it still does the same thing. Even if i switch browsers. Basically all the points i write above applies to the latest version.

**Comment by lastzero** [comment]
Strange, still working for me. Any special input devices or drivers in place that might change the event that is fired? From what I remember having read the code years ago, it won't fire if a mobile device is detected to avoid issues with touch gestures.

**Comment by lastzero** [comment]
Think it's because either iOS or Android use contextmenu to signal "long touch". Click is fired for a "short touch". The click duration may also play a role on desktop computers, it's worth testing at least. I know we have long click implemented in our search result views (list, mosaic, cards).

**Comment by Pheggas** [comment]
> Strange, still working for me. Any special input devices or drivers in place that might change the event that is fired? From what I remember having read the code years ago, it won't fire if a mobile device is detected to avoid issues with touch gestures.

Yeah. I mean, on my main PC. I tried same thing on older PC with no additional input devices attached. Same thing.

**Comment by Pheggas** [comment]
> Think it's because either iOS or Android use contextmenu to signal "long touch". Click is fired for a "short touch". The click duration may also play a role on desktop computers, it's worth testing at least. I know we have long click implemented in our search result views (list, mosaic, cards).

Might be. Definitely worth a try of deleting that chunk of code for the test build.

What i actually found out is, this doesn't happen on the demo site of PhotoPrism. Or, that I thought. Then I realized this issue only happens on 16:9 aspect ratio photos in landscape mode. On the other hand, in portrait mode with 9:16 aspect ratio, right click only zooms in the photo which is the expected result i think. 

So the next step would be upload test photo on demo site of PhotoPrism with variables that I described above and see what happen. 
In case it wouldn't show this issue, I would upload one of my photos from my personal gallery where I'm 100% sure it have this issue.

**Comment by lastzero** [comment]
Could it depend on the original or thumbnail pixel size? Context menu may be somehow coupled with the zoom function. Would be great if you can check the PhotoSwipe issue tracker for related problems 👍

Does your server use default thumb settings or did you change anything?

**Comment by Pheggas** [comment]
According to my settings, I'm pretty sure those are default ones.
![Screenshot_2021-11-14-07-56-28-893_com kiwibrowser browser](https://user-images.githubusercontent.com/9994511/141670852-41a4fef0-6cce-4dee-b6e3-595e8ed8639e.jpg)

**Comment by Pheggas** [comment]
I was able to replicate the issue inside your demo instance of Photoprism. In [this album](https://demo.photoprism.org/albums/ar2l2u915xokn8qm/newtestalbum), you can see two photos. One of which is from my personal gallery and the second one (with the lake thing), is from net. Both of these are 16:9 aspect ratio images and each have different resolutin.

This fact can approve my theory.

If the aspect ratio equals 16:9, you **CAN** face this issue.

**Comment by graciousgrey** [comment]
This seems to be fixed with the latest viewer improvements.
